src/golang/cmd/server/handler/connect_integration.go

```diff
@@ -159,27 +163,53 @@ func (h *ConnectIntegrationHandler) Perform(ctx context.Context, interfaceArgs i
 		return emptyResp, statusCode, err
 	}
 
-	if args.SetAsStorage {
-		// This integration should be used as the new storage layer
-		if err := setIntegrationAsStorage(
-			ctx,
-			args.Service,
-			args.Config,
-			h.WorkflowDagReader,
-			h.WorkflowDagWriter,
-			h.ArtifactReader,
-			h.ArtifactResultReader,
-			h.OperatorReader,
-			h.Database,
-		); err != nil {
-			return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unable to change metadata store.")
-		}
-	}
-
 	if err := txn.Commit(ctx); err != nil {
 		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unable to connect integration.")
 	}
 
+	if args.SetAsStorage {
+		// This integration should be used as the new storage layer.
+		// In order to do so, we need to migrate all content from the old store
+		// to the new store. This requires pausing the server and then restarting it.
+		// All of this logic is performed asynchronously so that the user knows that
+		// the connect integration request has succeeded and that the migration is now
+		// under way.
+		go func() {
+			// Wait until the server is paused
+			h.PauseServer()
```

Comment:
Are we pausing all scheduled workflows in `PauseServer()`?

Reply:
`PauseServer` can only pause HTTP requests. For the workflows, since that is a cron job we are running, we have to use a different mechanism. This is why we are acquiring the `ExecutionLock` in this function a few lines before.

At the start of each workflow run the executor now has to acquire a read lock for this mutex: see https://github.com/aqueducthq/aqueduct/blob/main/src/golang/cmd/executor/executor/workflow.go#L68. This allows us to block workflow runs.

Reply:
I see, so once this lock is acquired by the server, it will prevent any new workflow to be scheduled? In this case, what will happen to the workflow? I assume the migration time is short and newly scheduled workflow will eventually acquire the lock before the acquisition time-out?